lib/node_modules/@stdlib/math/generics/random/sample/README.md

```diff
@@ -87,15 +87,97 @@ out = sample( x, {
 The `probs` option __must__ be a numeric array consisting of non-negative values which sum to one. When sampling *without* replacement, note that the `probs` option denotes the initial element probabilities which are then updated after each draw.
 
 
-#### sample.factory( \[seed\] )
+#### sample.factory( \[pool, \]\[options\] )
 
-Returns a seeded `function` to sample elements from an `array`-like object.
+Returns a `function` to sample elements from an `array`-like object. 
 
 ``` javascript
-var mysample = sample.factory( 232 );
+var mysample = sample.factory();
 
 var out = mysample( [ 0, 1, 2, 3, 4 ] )
-// returns [ 4, 3, 4, 4 ]
+// e.g., returns [ 4, 3, 4, 4 ]
+```
+
+If provided an array-like object `pool`, the created function will be bound to sample from the supplied object.
+
+``` javascript
+var mysample = sample.factory( [1,2,3,4,5,6] );
+
+var out = mysample();
+// e.g., returns [ 2, 4, 1, 6, 5, 1 ]
+
+out = mysample();
+// e.g., returns [ 5, 2, 3, 6, 1, 4 ]
+```
+
+The function accepts the following `options`:
+
+* __seed__: pseudorandom number generator seed.
+* __size__: sample size. Default: `x.length=N`.
+* __replace__: `boolean` indicating whether to sample from `x` with replacement. Default: `true`.
+
+To seed the pseudorandom number generator, set the `seed` option.
+
+``` javascript
+var mysample = sample.factory({
+    'seed': 430
+});
+
+var out = mysample( [1,2,3,4,5,6] );
+// returns [ 1, 1, 1, 5, 4, 4 ]
+
+mysample = sample.factory( [1,2,3,4,5,6], {
+    'seed': 430
+});
+
+out = mysample();
+// returns [ 1, 1, 1, 5, 4, 4 ]
+```
+
+By default, the returned function draws samples of size `N`. To override the default `size`, set the `size` option.
+
+``` javascript
+var mysample = sample.factory({
+    'size': 4
+});
+
+var out = mysample( [0,1] );
+// e.g., returns [ 0, 0, 0, 1 ]
+
+// Default option can be overridden:
+out = mysample( [0,1], {
+    'size': 1
+});
+// e.g, returns [ 1 ]
+```
+
+By default, the returned function draws samples *with* replacement. To override the default `replace` strategy, set the `replace` option.
+
+``` javascript
+var mysample = sample.factory({
+    'replace': false
+});
+
+var out = mysample( [1,2,3] );
+// e.g., returns [ 3, 1, 2 ]
+```
+
+When fixing the population to sample from, the created function returns `null` after all population units are exhausted.
+
+```
+mysample = sample.factory( [1,2,3,4,5,6], {
```

Comment:
This `README.md` file is indeed not consistent with respect to variable declarations. However, the past we sometimes did not re-declare variables if they were already defined in one of the previous code blocks. Does that still hold?